girder_tech_journal/__init__.py

```diff
@@ -36,13 +36,52 @@
 from girder.models.user import User
 from girder.utility import mail_utils
 from girder.utility.model_importer import ModelImporter
+from girder.plugin import GirderPlugin
 from girder import events
 from girder_worker_utils.transforms.girder_io import GirderUploadToItem
 from tech_journal_tasks.tasks import processGithub, surveySubmission
 from . import constants
 from .models.journal import download_statistics, Journal
 
 
+class TechJournalPlugin(GirderPlugin):
+    DISPLAY_NAME = 'Tech Journal'
+    CLIENT_SOURCE_PATH = 'web_client'
+
+    def load(self, info):
+        techJournal = TechJournal()
+        techJournal.download_statistics = download_statistics()
+        techJournal.journal_collection = Journal()
+        ModelImporter.registerModel('journal', Journal, plugin='tech_journal')
+        mail_utils.addTemplateDirectory(os.path.join(
+            os.path.dirname(__file__),
+            'mail_templates'
+        ))
+
+        events.bind('model.setting.validate', 'journalMain', validateSettings)
+        info['apiRoot'].journal = techJournal
+        info['config']['/tech_journal'] = {
+            'tools.staticdir.on': True,
+            'tools.staticdir.dir': os.path.join(
+                os.path.abspath('girder-tech-journal-gui'), 'dist'),
+            'tools.staticdir.index': 'index.html'
+        }
+
+        # Bind REST events
+        events.bind('rest.get.folder/:id/download.after',
+                    'tech_journal',
+                    techJournal._onDownloadFileComplete)
+        events.bind('rest.get.journal/:id/details.after',
+                    'tech_journal',
+                    techJournal._onPageView)
+        events.bind('_queueEmails', 'tech_journal._queueEmails', _queueEmails)
```

Comment:
I'd move the `_queueEmails` and `sendEmails` to their own `utils.py` file, to prevent issues with circular imports. However, these probably should just be upstreamed to Girder itself too.